seqr/views/apis/family_api.py

```diff
@@ -45,9 +45,22 @@ def family_page_data(request, family_guid):
     sample_models = Sample.objects.filter(individual__family=family)
     samples = get_json_for_samples(sample_models, project_guid=project.guid, family_guid=family_guid, skip_nested=True, is_analyst=is_analyst)
     response = {
-        'samplesByGuid': {s['sampleGuid']: s for s in samples},
+        'samplesByGuid': {s['sampleGuid']: {**s, 'rnaSeqTypes': []} for s in samples},
     }
 
+    # Add Rna Seq metadata to samples
+    rna_seq_models = [
+        (RnaSeqTpm, 'TPM'),
+        (RnaSeqOutlier, 'Outlier'),
+        (RnaSeqSpliceOutlier, 'Splice Outlier'),
+    ]
+    for model, rna_seq_type in rna_seq_models:
+        rna_seq_samples = model.objects.filter(
+            sample__in=sample_models, sample__is_active=True
+        ).values('sample__guid').distinct()
```

Comment:
The more readable/maintainable way to do this would be to pass the `additional_values` argument into  `get_json_for_samples` with an `ArrayAgg` expression to get this list at query time instead of needing this post processing step

Reply:
I'm using `additional_values` instead but wasn't able to get `ArrayAgg` to work. I'm still annotating 3 fields and doing some post-processing in python to turn it into an array of strings.

Reply:
I tried an `ArrayAgg` with 3 cases statements but was seeing `function array_agg(text, text, text) does not exist`

Reply:
I think its also fine to just return 3 booleans and have the frontend handle that, the backend response does not *have* to be a single list of strings

Reply:
I was going to do it that way but then I was like.. I'd rather write the code to parse the 3 booleans into strings in python 😆

Reply:
I hear that, but the parsing code looks pretty messy in the end, and in general we try not to put display logic in the server side code and keep display logic in the UI. Also, doing the formatting when we are rendering skips this extra iteration we are running on the server side to add formatting, and instead does the formatting on the client as part of all the other formatting that is being done. I think the JS code would be really simple, something like - 
```
const RNA_TYPES =  [
  ('rnaSeqTpm', 'TPM'), ('rnaSeqOutlier', 'Expression Outlier'), ('rnaSeqSpliceOutlier', 'Splice Outlier),
]
...
{loadedSample.sampleType === SAMPLE_TYPE_RNA &&
   `RNAseq methods: ${RNA_TYPES.map(
       (field, title) => loadedSample[field] && title,
    ).filter(value => value).join(', ')}`}
```